finesse/hardware/plugins/temperature/senecat121.py

```diff
@@ -0,0 +1,141 @@
+    default_baudrate=57600,
+):
+    """An interface for Seneca T121 temperature readers.
+
+    The manual for this device is available at:
+    https://www.seneca.it/media/6848058/flyer_t12x_eng_nov_2020.pdf
+    """
+
+    def __init__(self, *serial_args: Any, **serial_kwargs: Any) -> None:
+        """Create a new Seneca from an existing serial device.
+
+        Args:
+            serial_args: Arguments to Serial constructor
+            serial_kwargs: Keyword arguments to Serial constructor
+        """
+        SerialDevice.__init__(self, *serial_args, **serial_kwargs)
+        TemperatureMonitorBase.__init__(self)
+
+    def close(self) -> None:
+        """Close the connection to the device."""
+        self.serial.close()
+
+    def read(self) -> bytes:
+        """Read temperature data from the Seneca.
+
+        Returns:
+            data: the sequence of bytes read from the device
+
+        Raises:
+            SenecaError: Malformed message received from device
+        """
+        try:
+            data = self.serial.read(size=21)
+        except SerialException as e:
+            raise SenecaError(e)
+
+        # require 21 bytes else checks will fail
+        min_length = 21
+        if len(data) != min_length:
+            raise SenecaError("Insufficient data read from device")
+
+        return data
+
+    def request_read(self) -> None:
+        """Write a message to the Seneca to prepare for a read operation.
+
+        Raises:
+            SenecaError: Error writing to the device
+        """
+        try:
+            self.serial.write(bytearray([1, 3, 0, 2, 0, 8, 229, 204]))
+        except Exception as e:
+            raise SenecaError(e)
+
+    def parse_data(self, data: bytes) -> list[Decimal]:
+        """Parse temperature data read from the Seneca.
+
+        The sequence of bytes is put through the conversion function and translated
+        into floats.
+
+        Args:
+            data: The bytes read from the device.
+
+        Returns:
+            vals: A list of Decimals containing the temperature values recorded
+                by the Seneca device.
+        """
+        ints = numpy.frombuffer(data, numpy.uint16, 8, 3)
+        print("ints:", ints, type(ints))
+
+        vals = [Decimal(float(self.calc_temp(val))) for val in ints]
+        return vals
+
+    def calc_temp(self, val: numpy.float64) -> numpy.float64:
+        """Convert data read from the Seneca device into temperatures.
+
+        Args:
+            val: A value from the array described by the data received from the device.
+
+        Returns:
+            vals: The converted value.
+        """
+        temp = (self.range * ((val / 1000) - self.min_volt)) + self.min_temp
+        print("temp:", temp, type(temp))
+        return temp
+
+    def get_temperatures(self) -> list[Decimal]:
+        """Get the current temperatures."""
+        self.request_read()
+        data = self.read()
+        print("range:", self.range)
```

Comment:
```suggestion
```

Presumably this was just for testing?

Reply:
Ah yep, well spotted. I'll get rid of those.